src/main/resources/org/jenkinsci/plugins/workflow/job/WorkflowRun/sidepanel.jelly

```diff
@@ -34,6 +34,37 @@
             <p:console-link/>
             <l:task icon="images/24x24/notepad.png" href="${buildUrl.baseUrl}/configure" title="${h.hasPermission(it,it.UPDATE)?'%Edit Build Information':'%View Build Information'}"/>
             <st:include page="delete.jelly"/>
+            <j:if test="${it.isBuilding() and h.hasPermission(it,it.CANCEL)}">
+                <j:choose>
+                    <j:when test="${it.hasStopBeenCalled() == false}">
+                        <l:task icon="images/24x24/edit-delete.png"
+                                href="${buildUrl.baseUrl}/stop"
+                                title="${%Stop Build}"
+                                contextMenu="false"
+                                post="true"
+                                requiresConfirmation="true"
+                                confirmationMessage="Really stop build?"/>
+                    </j:when>
+                    <j:when test="${it.hasTermBeenCalled() == false}">
+                        <l:task icon="images/24x24/edit-delete.png"
+                                href="${buildUrl.baseUrl}/term"
+                                title="${%Abort Running Steps}"
+                                contextMenu="false"
+                                post="true"
+                                requiresConfirmation="true"
+                                confirmationMessage="Really forcibly abort running steps?"/>
+                    </j:when>
+                    <j:when test="${it.hasKillBeenCalled() == false}">
```

Comment:
Also the current UI allows you to try `stop` multiple times without using `term`, or try `term` multiple times without using `kill`. Each `stop` will send a polite signal to the running step(s); even when the first signal did not suffice, a second might. Each `term` will immediately halt the running step(s); even if stopping that step did not halt the build, stopping a _later_ step might. For example:

```groovy
catchError {
  someHangingStepThatCannotBeInterrupted()
}
mail …
```

If you `stop`, nothing will happen, and after 15s you get the option to `term`. Say you take it, killing the hanging step. Now the build proceeds to the `mail` step. Say this in turn takes too long, perhaps trying to contact a unresponsive SMTP server. A simple `stop` should suffice to kill `mail` politely. You do not need `term` or `kill`.

Basically we want to avoid escalating unless and until it is really necessary.